File: server/models/subject_schedule.go
package models

import (
	surrealModels "github.com/surrealdb/surrealdb.go/pkg/models"
)

type SubjectScheduleEntity struct {
	ID        surrealModels.RecordID `json:"id" validate:"required"  swaggertype:"object"`
	DayOfWeek int                    `json:"day_of_week" validate:"required,min=0,max=6"`

	StartingHour   int `json:"starting_hour" validate:"required,min=0,max=23"`
	StartingMinute int `json:"starting_minute" validate:"required,min=0,max=59"`

	EndingHour   int `json:"ending_hour" validate:"required,min=0,max=23"`
	EndingMinute int `json:"ending_minute" validate:"required,min=0,max=59"`

	SubjectOffer surrealModels.RecordID `json:"subject_offer" validate:"required" swaggertype:"object"`
}

File: client/src/interfaces/SubjectSchedule.ts
import type { Id } from './surrealDb';

export interface SubjectSchedule {
  id: Id;

  starting_hour: number;
  starting_minute: number;

  ending_hour: number;
  ending_minute: number;

  day_of_week: 0 | 1 | 2 | 3 | 4 | 5 | 6; // Represents days of the week from Sunday (0) to Saturday (6)

  subject_section: Id;
}
